### PR TITLE
[Model] Add PaliGemma2 (SigLIP + Gemma2) with LoRA support

### PR DESCRIPTION
**Pull request by shichiachi3-cyber**
## Motivation

Add support for [PaliGemma 2](https://huggingface.co/google/paligemma2-3b-pt-448), Google's open-weight vision-language model family (3B / 10B / 28B; 224 / 448 / 896 resolutions). PaliGemma 2 is widely used for industrial vision, document understanding, and small-footprint VLM fine-tuning, and is a natural fit for SGLang since both of its sub-models are already supported:

- `SiglipVisionModel` — already in SGLang (`python/sglang/srt/models/siglip.py`)
- `Gemma2ForCausalLM` — already in SGLang (`python/sglang/srt/models/gemma2.py`)

This PR is essentially a 266-line wiring layer between the two, plus LoRA hooks.

## Modifications

Single new file: `python/sglang/srt/models/paligemma.py`

Architecture:

```
SiglipVisionModel  →  PaliGemmaMultiModalProjector (linear)  →  Gemma2ForCausalLM
   (existing)               (new, ~12 lines)                       (existing)
```

Key implementation details:

- **Linear projector** (not Gemma3's pooling projector) — PaliGemma 2 uses a single `nn.Linear(vision_hidden_size, projection_dim)`.
- **Embedding scaling** — `projected * (hidden_size ** -0.5)` matches the HuggingFace reference.
- **Image-token padding** — uses `MultiModalityDataPaddingPatternMultimodalTokens` (no start/end pair, just repeated `<image>` placeholders).
- **Weight remapping** — handles the transformers ≥ 4.52 prefix change (`model.language_model.*` → `language_model.*`, `lm_head.*` → `language_model.lm_head.*`).
- **Auto-registration** — `EntryClass = PaliGemmaForConditionalGeneration` at file end; no other registry changes needed.

LoRA support:

- `supports_lora = True`
- `supported_lora_modules = ["qkv_proj", "o_proj", "gate_up_proj", "down_proj"]`
- `lora_pattern` regex restricts adapters to **language model layers only** — vision tower and projector are excluded:

```python
lora_pattern = re.compile(
    r"^language_model\.model\.layers\.\d+\."
    r"(?:self_attn|mlp)\."
    r"(?:qkv_proj|o_proj|gate_up_proj|down_proj)"
)
```

This matches PEFT's default target-module convention for PaliGemma 2 fine-tunes.

## Usage

```bash
# Base model
python -m sglang.launch_server \
  --model-path google/paligemma2-3b-pt-448 \
  --dtype bfloat16 \
  --host 0.0.0.0 --port 30000

# With one or more LoRA adapters
python -m sglang.launch_server \
  --model-path google/paligemma2-3b-pt-448 \
  --dtype bfloat16 \
  --lora-paths my-defect-adapter=/path/to/peft/adapter \
  --max-loras-per-batch 2 \
  --host 0.0.0.0 --port 30000
```

OpenAI-compatible request:

```bash
curl http://127.0.0.1:30000/v1/chat/completions \
  -H "Content-Type: application/json" \
  -d '{
    "model": "google/paligemma2-3b-pt-448",
    "messages": [{"role": "user", "content": [
      {"type": "image_url", "image_url": {"url": "https://example.com/board.jpg"}},
      {"type": "text", "text": "describe defects on this PCB"}
    ]}]
  }'
```

## Accuracy Tests

Tested against `google/paligemma2-3b-pt-448` with PEFT-trained LoRA adapters on industrial PCB defect detection (6 adapters: missing_hole, mouse_bite, open_circuit, short, spur, spurious_copper). Generation outputs match the HuggingFace `transformers` reference on the same prompts.

A formal VLM benchmark (MMMU, MathVista, etc.) on the base model has not been included in this PR — happy to add one if reviewers prefer; suggestions on which suite the project considers canonical for VLM additions are welcome.

## Speed Tests and Profiling

Not measured. The model is a thin SigLIP encode + linear projector on top of Gemma2 decoding, so end-to-end throughput should track Gemma2 closely with a fixed per-image vision overhead. Happy to run `bench_serving.py` numbers if reviewers want them in this PR.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests). *(Manual smoke test only; happy to add a unit test under `test/srt/models/` — please point me at the canonical fixture pattern for VLM additions.)*
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations). *(Will add a docs entry under `docs/supported_models/` once a maintainer confirms preferred format.)*
- [ ] Provide accuracy and speed benchmark results. *(See sections above — qualitative parity verified; formal benchmarks pending reviewer guidance.)*
- [x] Follow the SGLang code style guidance.

### CODE REVIEW

**Comment by shichiachi3-cyber** [comment]
Hi, this PR adds PaliGemma2 support with multi-LoRA serving to SGLang.  The Gemini review feedback has been addressed (dtype fix, load_weights delegation, weight key remapping, redundant stacked_params cleanup).  
                                                                                                               
Could a maintainer please approve the CI gate so the test suite can run?

Happy to address any further review comments. Thank you!

**Comment by shichiachi3-cyber** [comment]
Heads up — this PR is currently failing CI because the `pr-gate` workflow requires a `run-ci` label per the [contribution guide](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests).

Could a maintainer please add the label when convenient? The 8 red checks are all the same `Require run-ci label (optional)` step gating downstream test runs — once the label is added, real CI will exercise the change.

Pre-commit (isort / ruff / black / codespell) was re-run locally on `python/sglang/srt/models/paligemma.py` against current `main` and passes cleanly, so I expect the actual test pipeline to be in good shape once it can run.
